src/backend/mpi/swdsm.cpp

```diff
@@ -95,22 +93,20 @@ char * barwindowsused;
 sem_t ibsem;
 
 /*Loading and Prefetching*/
-/** @brief Tracking address of pages that should be loaded by loadthread1 */
-unsigned long *loadline;
-/** @brief Tracking cacheindex of pages that should be loaded by loadthread1 */
-unsigned long *loadtag;
-/** @brief Tracking address of pages that should be loaded by loadthread2 */
-unsigned long *prefetchline;
-/** @brief Tracking cacheindex of pages that should be loaded by loadthread2 */
-unsigned long *prefetchtag;
-/** @brief loadthread1 waits on this to start loading remote pages */
-sem_t loadstartsem;
-/** @brief signalhandler waits on this to complete a transfer */
-sem_t loadwaitsem;
-/** @brief loadthread2 waits on this to start loading remote pages */
-sem_t prefetchstartsem;
-/** @brief signalhandler waits on this to complete a transfer */
-sem_t prefetchwaitsem;
+/**
+ * @brief load into cache helper function
+ * @param tag aligned address to load into the cache
+ * @param line cache entry index to use
+ */
+void load_cache_entry(unsigned long tag, unsigned long line);
+
+/**
+ * @brief prefetch into cache helper function, which duplicates a lot of code
+ * @param tag aligned address to prefetch into the cache
+ * @param line cache entry index to use
+ * @todo this function duplicates a lot of code from load_cache_entry(), this should be fixed
```

Comment:
I would have preferred for this to have been fixed here, but at least please open a ticket for it.

Reply:
The difference between `load_cache_entry` and `prefetch_cahce_entry` is actually extremely minimal. The only functional difference I can find between them is that the former performs the test `tempsharer != id` in one of the if-cases, while the later does not perform the test in the equivalent if-case.

Reply:
related to #28?

Reply:
I've been looking a bit at the relevant if-statement. My conclusion is that the `tempsharer != id` condition is not strictly necessary, the side-effects of the if-case will be skipped anyway should the condition be true. Someone should double check this however.

Reply:
This issue is now tracked in #32